core/build/build.go

```diff
@@ -0,0 +1,24 @@
+package build
+
+// The build module utilizes build tags to determine the environment that this binary was built to target
+// the currently supported build modes are dev, test. Setting both tags is not allowed and will result to compilation errors.
+
+type Build string
+
+const (
+	Prod Build = "prod"
+	Dev  Build = "dev"
+	Test Build = "test"
+)
```

Comment:
Does defining this `Build` type add value? Seems like pure string would work the same, especially if s/`Mode`/`mode`:
```suggestion
const (
	Prod = "prod"
	Dev  = "dev"
	Test = "test"
)
```